process/coordinator/process.go

```diff
@@ -489,10 +523,19 @@ func (tc *transactionCoordinator) CreateMbsAndProcessTransactionsFromMe(
 		miniBlocks = append(miniBlocks, interMBs...)
 	}
 
-	rewardsMBs := tc.createRewardsMiniBlocks()
-	if len(rewardsMBs) > 0 {
-		miniBlocks = append(miniBlocks, rewardsMBs...)
+	rewardMb := tc.createRewardsMiniBlocks()
+	if len(rewardMb) == 0 {
+		log.Error("could not create reward mini-blocks")
+	}
+
+	rewardsPreProc := tc.getPreProcessor(block.RewardsBlock)
```

Comment:
when creating the reward miniblock, the intermediate processor could also update the state, process effectivele the rewards.

Reply:
moved this away from tx coordinator